### PR TITLE
fix(catalog): remove timeout and add logs

### PR DESCRIPTION
**Pull request by alexcodelf**
**Problem:**
- Sync an template with tags more than 100 may take more than 5 min to update template version, remove timeout.
- The logs of sync an catalog are too brief, it is hard to monitor the progress of sync an catalog.

The reason is the cron task will set timeout with previous task time. While syncing catalog, the timeout limit will be longer than the calculation after setting a short crontab expression, it may lead to timeout.

**Solution:**
- Remove timeout of sync template.
- Add debug logs of sync an catalog.
- Use a longer task timeout, using a new context to update catalog status.

**Related Issue:**

#1048

### CODE REVIEW

File: pkg/templates/git.go [hunk 1]
```diff
@@ -431,15 +432,16 @@ func getValidVersions(
 			Mode:   git.HardReset,
 		})
 		if err != nil {
-			logger.Warnf("failed to reset to tag %s: %v", tag, err)
+			logger.Warnf("failed set to %s/%s: %v", templateName, tag, err)
```
Comment (thxCode) [inline pkg/templates/git.go:435]: failed to set %s/%s

File: pkg/templates/git.go [hunk 2]
```diff
@@ -286,7 +286,7 @@ func SyncTemplateFromGitRepo(
 	defer func() {
 		rerr := updateTemplateStatus(ctx, mc, entity, err)
 		if rerr != nil {
-			logger.Errorf("failed to update template status: %v", rerr)
+			logger.Errorf("failed to update template %s status: %v", entity.Name, rerr)
```
Comment (thxCode) [inline pkg/templates/git.go:289]: id is better than name.

**Review by thxCode — COMMENTED** [review]
can we unify the info in the format: `"catalog.Name/template.Name:template.Version"`?

**Comment by alexcodelf** [comment]
> can we unify the info in the format: `"catalog.Name/template.Name:template.Version"`?

template may not belongs to a catalog.

**Comment by thxCode** [comment]
> > can we unify the info in the format: `"catalog.Name/template.Name:template.Version"`?
> 
> template may not belong to a catalog.

use "template.Name:template.Version"

File: pkg/catalog/catalog.go [hunk 3]
```diff
@@ -135,15 +142,22 @@ func SyncTemplates(ctx context.Context, mc model.ClientSet, c *model.Catalog) er
...
-						fmt.Errorf("error syncing template %s: %w",
-							repo.Name, serr))
+						fmt.Errorf("error syncing \"%s:%s\" of catalog %q: %w",
+							c.Name, repo.Name, c.ID, serr))
+
+					atomic.AddInt32(&failed, 1)
+				} else {
+					atomic.AddInt32(&processed, 1)
 				}
+
```
Comment (aiwantaozi) [inline pkg/catalog/catalog.go:158]: total: %d

File: pkg/templates/git.go (hunk 1 above)
Comment (alexcodelf) [reply pkg/templates/git.go]: update to %s:%s

**Review by aiwantaozi — APPROVED** [review]

File: staging/utils/cron/cron.go [hunk 4]
```diff
@@ -216,19 +216,23 @@ func (in *scheduler) Schedule(jobName string, cron Expr, task Task, taskArgs ...
...
+		next = ceParsed.Next(now).Sub(now) * 3
 	)
 
-	if next > atLeast {
-		next >>= 1
-	}
-
-	if next < atLeast {
+	switch {
+	case next < 0:
```
Comment (thxCode) [inline staging/utils/cron/cron.go:232]: next <= 0 ?

**Review by aiwantaozi — APPROVED** [review]

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**